test/controllers/supplementary/TransportInformationContainersPageControllerSpec.scala

```diff
@@ -0,0 +1,269 @@
+        )
+        withCaching[TransportInformationContainerData](Some(cachedData), id)
+
+        val body = removeActionURLEncoded("0")
+        val result = route(app, postRequestFormUrlEncoded(uri, body)).get
+
+        status(result) must be(SEE_OTHER)
+      }
+    }
+
+    "display the form page with an error" when {
+
+      "try to add a container without any data" in {
+
+        withCaching[TransportInformationContainerData](None, id)
+        val body = Seq(("id", ""), addActionURLEncoded)
+
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+        val stringResult = contentAsString(result)
+
+        status(result) must be(BAD_REQUEST)
+        stringResult must include(messages("supplementary.transportInfo.containerId.empty"))
+      }
+
+      "try to save and continue without any data" in {
+
+        val body = Seq(("id", ""), saveAndContinueActionURLEncoded)
+
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+        val stringResult = contentAsString(result)
+
+        status(result) must be(BAD_REQUEST)
+        stringResult must include(messages("supplementary.continue.mandatory"))
+      }
+
+      "try to add a longer id" in {
+
+        val body = Seq(("id", createRandomString(20)), addActionURLEncoded)
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+
+        status(result) must be(BAD_REQUEST)
+        contentAsString(result) must include(messages("supplementary.transportInfo.containerId.error"))
+      }
+
+      "try to add non-alphanumeric characters" in {
+
+        val body = Seq(("id", "<123wqeds>"), addActionURLEncoded)
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+
+        status(result) must be(BAD_REQUEST)
+        contentAsString(result) must include(messages("supplementary.transportInfo.containerId.error"))
+      }
+
+      "try to add duplicated container" in {
+
+        val cachedData = TransportInformationContainerData(Seq(TransportInformationContainer("M1l3s")))
+        withCaching[TransportInformationContainerData](Some(cachedData), id)
+
+        val body = Seq(("id", "M1l3s"), addActionURLEncoded)
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+
+        status(result) must be(BAD_REQUEST)
+        contentAsString(result) must include(messages("supplementary.duplication"))
+      }
+
+      "try to add more than 9999 containers" in {
+        withCaching[TransportInformationContainerData](Some(cacheWithMaximumAmountOfHolders), id)
+
+        val body = Seq(("id", "M1l3s"), addActionURLEncoded)
+        val result = route(app, postRequestFormUrlEncoded(uri, body: _*)).get
+
+        status(result) must be(BAD_REQUEST)
+        contentAsString(result) must include(messages("supplementary.limit"))
+      }
+
+      "try to save more than 9999 containers" in {
```

Comment:
what's the difference between this and the above test in 182?

Reply:
One `adds` and the other `saves and continues`